geomesa-datastore-bundle/geomesa-datastore-processors/src/main/scala/org/geomesa/nifi/datastore/processor/geotools/GeoToolsProcessor.scala

```diff
@@ -22,9 +21,9 @@ abstract class GeoToolsProcessor extends DataStoreProcessor(Seq.empty) {
 
   private var dataStoreName: PropertyDescriptor = _
 
-  override protected def init(context: ProcessorInitializationContext): Unit = {
-    dataStoreName = GeoToolsProcessor.dataStoreName(listDataStores().map(_.getDisplayName).toSeq)
-    super.init(context)
+  override def reloadDescriptors(): Unit = {
+    dataStoreName = GeoToolsProcessor.dataStoreName(GeoToolsProcessor.listDataStores().map(_.getDisplayName).toSeq)
+    super.reloadDescriptors()
```

Comment:
Neat.  When does reloadDescriptors get called?

Reply:
onAdded is the annotation, also in the protected AbstractProcessor.init method. but seems to be called at least when the properties are updated (or maybe only properties that are marked as modifying the classpath)